packages/headless/src/features/facets/automatic-facet-set/automatic-facet-set-slice.ts

```diff
@@ -48,6 +51,38 @@ export const automaticFacetSetReducer = createReducer(
         for (const value of facet.values) {
           value.state = 'idle';
         }
+      })
+      .addCase(restoreSearchParameters, (state, action) => {
+        const af = action.payload.af ?? {};
+
+        for (const field in af) {
+          const response = buildTemporaryAutomaticFacetResponse(field);
+          const values = af[field].map((value) =>
+            buildTemporarySelectedFacetValue(value)
+          );
+          response.values.push(...values);
+
+          state.facets[field] = response;
+        }
       });
   }
 );
+
+function buildTemporaryAutomaticFacetResponse(
+  field: string
+): AutomaticFacetResponse {
+  return {
+    field,
+    values: [],
+    moreValuesAvailable: false,
+    label: '',
+    indexScore: 0,
+  };
+}
+function buildTemporarySelectedFacetValue(value: string): FacetValue {
```

Comment:
nit: new line before function def
didn't the lint/prettier catch/fix this?

Reply:
Thanks for reviewing again, I dont think we have a rule for this. I can look into how to add a rule for this case as a technical debt.